client/lib/url/decode-utils.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Internal dependencies
+ */
+import { URL as TypedURL } from 'types';
+
+interface Stringable {
+	toString: () => string;
+}
+
```

Comment:
It's a bit tempting to make a wrapper for both of these functions given their bodies are duplicated except for the function the call on the treated input 🙂